src/V3EmitCImp.cpp

```diff
@@ -113,46 +113,61 @@ class EmitCImp final : EmitCFunc {
         if (!first) puts("\n");
     }
     void emitCtorImp(const AstNodeModule* modp) {
-        const string modName = EmitCUtil::prefixNameProtect(modp);
+        const std::string modName = EmitCUtil::prefixNameProtect(modp);
 
         puts("\n");
         m_lazyDecls.emit("void " + modName + "__", protect("_ctor_var_reset"),
                          "(" + modName + "* vlSelf);");
         puts("\n");
 
-        putns(modp, modName + "::" + modName + "(" + EmitCUtil::symClassName()
-                        + "* symsp, const char* v__name)\n");
-        puts("    : VerilatedModule{v__name}\n");
+        const std::string ctorArgs = EmitCUtil::symClassName() + "* symsp, const char* namep";
 
-        ofp()->indentInc();
-        for (const AstNode* nodep = modp->stmtsp(); nodep; nodep = nodep->nextp()) {
-            if (const AstVar* const varp = VN_CAST(nodep, Var)) {
-                if (const AstBasicDType* const dtypep
-                    = VN_CAST(varp->dtypeSkipRefp(), BasicDType)) {
-                    if (dtypep->keyword().isMTaskState()) {
-                        puts(", ");
-                        putns(varp, varp->nameProtect());
-                        puts("(");
-                        iterateConst(varp->valuep());
-                        puts(")\n");
-                    } else if (varp->isIO() && varp->isSc()) {
-                        puts(", ");
-                        putns(varp, varp->nameProtect());
-                        puts("(");
-                        putsQuoted(varp->nameProtect());
-                        puts(")\n");
-                    } else if (dtypep->isDelayScheduler()) {
-                        puts(", ");
-                        putns(varp, varp->nameProtect());
-                        puts("{*symsp->_vm_contextp__}\n");
+        // The root module needs a proper constuctor, everything else uses a
+        // 'ctor' function in order to be able to split up constructors
+        if (modp->isTop()) {
+            putns(modp, modName + "::" + modName + "(" + ctorArgs + ")\n");
+
+            ofp()->indentInc();
+            const char* sepp = "  : ";
```

Comment:
I don't think two spaces matches what clang-format would do.

```suggestion
            const char* sepp = " : ";
```